### DISCUSSION TITLE
How to add another PHP version using Quick add

### DISCUSSION THREAD

**Discussion by Pen-y-Fan** [leokhoa/laragon#807] [Laragon's Articles & How-tos]
The [\[Tutorial\] How to add another PHP version](https://github.com/leokhoa/laragon/wiki/%5BTutorial%5D-How-to-add-another-PHP-version-(PHP-8.3.4)) has been updated with information how to use Laragon's **Quick add** for another PHP version.

I hope this helps.

**Comment by Antmans0000** [comment]
![image](https://github.com/user-attachments/assets/cfa44491-ea84-4816-908b-9b7ee510e716)
apache does not start (php8.3.9)

**Reply by Pen-y-Fan** [reply]
The PHP directory doesn't look right, /php-8.3.9.tar/ should be like /php-8.3.8-nts-Win32-vs16-x64/

Are you using a matched version of PHP and Apache? The both need to be the same architecture.

See my blog [How to update the PHP version in Laragon](https://pen-y-fan.github.io/2023/01/15/how-to-update-the-php-version-in-laragon/) for more details.

Edit: Updated the PHP directory.

**Reply by Antmans0000** [reply]
Thank you ! it works now
i tried to download from https://windows.php.net/download/ but the link doesn't work, 
i found this one https://phpdev.toolsforresearch.com/php-8.3.8-nts-Win32-vs16-x64.zip

Thank you again ! Have a good day ^^

**Comment by Sunchock** [comment]
I noticed that manual method uses Thread Safe version while Quick-add method uses NTS (Non Thread Safe) version.
What is the difference between the two ?
Does Laragon work the same for both methods ?

**Reply by Pen-y-Fan** [reply]
It's to do with compatibility with IIS, Apache and Nginix. See my blog [How to update the PHP version in Laragon](https://pen-y-fan.github.io/2023/01/15/how-to-update-the-php-version-in-laragon/) for more details.

**Comment by ghost** [comment]
f, like me, you work with several php versions (localhost) then create a .shwitch-php.bat to change the php version. Works wonderfully for me under Win 11 and Linux

**Comment by Pen-y-Fan** [comment]
ℹ️ [\[Tutorial\] How to add another PHP version](https://github.com/leokhoa/laragon/wiki/%5BTutorial%5D-How-to-add-another-PHP-version-(PHP-8.3.4)) updated with Instructions on how to add PHP 8.4 NTS using quick add. 

Apache:

![image](https://github.com/user-attachments/assets/24799392-7dbb-468e-915e-be5904fad6fe)

Nginx:

![image](https://github.com/user-attachments/assets/a9b46307-bfc5-4cf2-89cd-620efc28e34d)

Laragon:

![image](https://github.com/user-attachments/assets/bc1c0eea-7c44-4167-a4ed-801d9871eaaf)

⚠️The most recent version of PHP 8.4 is built with VS17 (Visual Studio 2022 compiler) for the best compatibility and performance upgrade to Apache VS17 too. See [\[Tutorial\] How to add another Apache version](https://github.com/leokhoa/laragon/wiki/%5BTutorial%5D-How-to-add-another-Apache-version) for the manual and quick add instructions.

**Reply by AWqxKAWERbXo** [reply]
awesome

**Reply by athomft** [reply]
> ℹ️ [[Tutorial] How to add another PHP version](https://github.com/leokhoa/laragon/wiki/%5BTutorial%5D-How-to-add-another-PHP-version-(PHP-8.3.4)) updated with Instructions on how to add PHP 8.4 NTS using quick add.
> 
> Apache:
> 
> ![image](https://private-user-images.githubusercontent.com/40126936/388667545-24799392-7dbb-468e-915e-be5904fad6fe.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYyMjk5NzQsIm5iZiI6MTczNjIyOTY3NCwicGF0aCI6Ii80MDEyNjkzNi8zODg2Njc1NDUtMjQ3OTkzOTItN2RiYi00NjhlLTkxNWUtYmU1OTA0ZmFkNmZlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTA3VDA2MDExNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI4OWZhOGJhZjA1ZGJiODgzODViNmZkODk4NzQ2YWI5MDE4Yjc3Zjg2NGUwNDBkMWEzMzhkYjgwM2M4ZDc3OWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.qJItSBRWIFWIvuSUxjRCwRbfshzZP9_xSd9bdWzlgXQ)
> 
> Nginx:
> 
> ![image](https://private-user-images.githubusercontent.com/40126936/388668988-a9b46307-bfc5-4cf2-89cd-620efc28e34d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYyMjk5NzQsIm5iZiI6MTczNjIyOTY3NCwicGF0aCI6Ii80MDEyNjkzNi8zODg2Njg5ODgtYTliNDYzMDctYmZjNS00Y2YyLTg5Y2QtNjIwZWZjMjhlMzRkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTA3VDA2MDExNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIwNjBmODZhOGZkOGE3NWY5ZGY0M2E0Nzc5ZWQ0Yjg1ZTExYjgzMjMxZWIxMDY3NjY2MmVlMDVkYzg4MWZmNjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.XpBvimEQziXMZWkrL1Kd43QHGvT7bpsT5RaUR8_O6Cs)
> 
> Laragon:
> 
> ![image](https://private-user-images.githubusercontent.com/40126936/388667814-bc1c0eea-7c44-4167-a4ed-801d9871eaaf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYyMjk5NzQsIm5iZiI6MTczNjIyOTY3NCwicGF0aCI6Ii80MDEyNjkzNi8zODg2Njc4MTQtYmMxYzBlZWEtN2M0NC00MTY3LWE0ZWQtODAxZDk4NzFlYWFmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTA3VDA2MDExNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY0NzkzYWFiOWZiMTM0YTE4NmEwMzM2ZTQ3M2U2MDVhYzU4N2E0NWRmNzA3NjY4MTkyZGE5NmI2N2U2ZWE4OTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.68JLMcSVyfFppBNECjZy52Pi6-auY2fwK-iQzGQRF8c)
> 
> ⚠️The most recent version of PHP 8.4 is built with VS17 (Visual Studio 2022 compiler) for the best compatibility and performance upgrade to Apache VS17 too. See [[Tutorial] How to add another Apache version](https://github.com/leokhoa/laragon/wiki/%5BTutorial%5D-How-to-add-another-Apache-version) for the manual and quick add instructions.

Hello @Pen-y-Fan,

I have successfully installed Apache 2.4.62 and PHP 8.4. However, I am encountering an issue with phpMyAdmin. I am unable to access or open phpMyAdmin.

Could you please assist me in resolving this issue?

**Reply by Pen-y-Fan** [reply]
I tent to use Heidi as switching PHP versions affects phpMyAdmin. There are some options here:

https://github.com/leokhoa/laragon/discussions/693#discussioncomment-8518456
